winning and loosing sounds and screens

xhelp00 opened the pull request.

Commit:
Working on items

Commit:
Hope its doesn't segfault anymore

AntessTerra committed:
Winning condition WIP

AntessTerra committed:
Added item hud, and toggle to turn the hud off f8
Also animation for items on pedestals

AntessTerra committed:
Added toggle in header to disable mouse control, in case of home testing

xhelp00 committed:
added sound of key and winning with delayed closing

Commit:
Changed a few stuff

xhelp00 committed:
winning texture

xhelp00 committed:
blending red image while being hit

Commit:
Fix to not die in end screen

Commit:
Fix of a memory leak

Commit:
Death note

xhelp00 committed:
loosing texture and sound added

xhelp00 committed:
added fun map hell.cub

xhelp00 commented:
blackhole texture added and temporarily is replacing floor and ceiling - check and decide if you want to keep it :smile:

xhelp00 committed:
fix

xhelp00 committed:
almost norm on main.c

xhelp00 committed:
norm main.c

xhelp00 committed:
norm sound.c

xhelp00 committed:
norm minimap.c

xhelp00 committed:
textures loading split

xhelp00 committed:
norm values.c

xhelp00 committed:
hud drawing split

xhelp00 committed:
norm draw_image.c

xhelp00 committed:
norm parser wip

xhelp00 committed:
norm parser.c

xhelp00 committed:
norm hook.c

xhelp00 committed:
norm cub3d.h

xhelp00 committed:
norm movement wip

xhelp00 committed:
refactoring movement wip

AntessTerra commented on draw_image.c:
```diff
@@ -74,140 +94,199 @@ int	extract_color(unsigned char *pixel)
+		box->info.color = extract_color(&box->textures[UI_PICKUPS].addr[(x - 4
+					- ((x - 20) / 2)) * 4 + box->textures[UI_PICKUPS].line_len
+				* (y - 50 - ((y - 50) / 2))]);
+}
```
just add this line here, cause the key counter icon didn't show up on screen
if ((box->info.color & 0x00FFFFFF) != 0)
		my_mlx_pyxel_put(&box->image, x, y, box->info.color);

AntessTerra replied:
Other wise, great job on norminetting (👉ﾟヮﾟ)👉

xhelp00 committed:
refactoring movement wip

xhelp00 committed:
wip

xhelp00 committed:
norm movement.c

xhelp00 committed:
floor cast split

xhelp00 committed:
casting almost normed

Commit:
Minor fixes of winning and losing

xhelp00 committed:
fucking norm

Commit:
Continue on minor fixing

Commit:
Fixed Spawning orientation and Larry and leech now can sleep

xhelp00 committed:
NORMED

xhelp00 committed:
additional map checks

xhelp00 committed:
fix for check around map

xhelp00 committed:
fix for check and menu redesign

xhelp00 committed:
map without sprites not crashing anymore

added check to ensure that the string returned by get_next_line is not only non-NULL but also non-empty before attempting to access its chars

Commit:
Started work on freeing stuff

Commit:
Returning code to hook_handle.c after wrong merge :{

Commit:
Possibly all leaks from our side taken care of, just need to do some testing on it

xhelp00 committed:
Libft update

xhelp00 committed:
music fixed

Commit:
Extendet range
